### ISSUE TITLE
Unknown procedure `to_list/2`

### ISSUE THREAD

**Issue by ogoletti** [SWI-Prolog/swipl-devel#963]
Hi, 

I couldn't run some plunit tests on 8.4.2-1 that I could on 8.4.1-1 (I tried with homebrew install and with package retrieved from https://www.swi-prolog.org/download/stable/bin/swipl-8.4.1-1.x86_64.dmg and https://www.swi-prolog.org/download/stable/bin/swipl-8.4.2-1.fat.dmg) 

I am on my M1 mac with MacOS Monterey 12.3 also tried on 12.2.1 intel

When running `?- run_tests(sql).` with 8.4.2 I get an error for missing `to_list/2`

Here is the stacktrace : 

```ERROR: Unknown procedure: '$messages':to_list/2
ERROR: In:
ERROR:   [38] to_list([selec/4],_6988)
ERROR:   [37] append_args([selec/4],[],_7028) at /opt/homebrew/Cellar/swi-prolog/8.4.2/libexec/lib/swipl/boot/messages.pl:2137
ERROR:   [34] actions_to_format([nl,...|...],_7066,_7068) at /opt/homebrew/Cellar/swi-prolog/8.4.2/libexec/lib/swipl/boot/messages.pl:2073
ERROR:   [33] actions_to_format(['  However, there are definitions for:',nl|...],_7106,_7108) at /opt/homebrew/Cellar/swi-prolog/8.4.2/libexec/lib/swipl/boot/messages.pl:2100
ERROR:   [32] actions_to_format([nl,'  However, there are definitions for:'|...],_7146,_7148) at /opt/homebrew/Cellar/swi-prolog/8.4.2/libexec/lib/swipl/boot/messages.pl:2073
ERROR:   [31] actions_to_format(['Unknown procedure: ~q'- ...,nl|...],_7186,_7188) at /opt/homebrew/Cellar/swi-prolog/8.4.2/libexec/lib/swipl/boot/messages.pl:2090
ERROR:   [30] actions_to_format(['~q:~q/~w: '- ...,...|...],_7232,_7234) at /opt/homebrew/Cellar/swi-prolog/8.4.2/libexec/lib/swipl/boot/messages.pl:2090
ERROR:   [29] message_to_string(error(existence_error(procedure,...),context(...,_7296)),_7278) at /opt/homebrew/Cellar/swi-prolog/8.4.2/libexec/lib/swipl/boot/messages.pl:2063
ERROR:   [28] plunit:failure(error(existence_error(procedure,...),context(...,_7348)),_7328,_7330) at /opt/homebrew/Cellar/swi-prolog/8.4.2/libexec/lib/swipl/library/plunit.pl:1778
ERROR:   [27] plunit:message(plunit(failed(sql,insert2,78,...)),[url(...),':\n\t'|...],_7382) at /opt/homebrew/Cellar/swi-prolog/8.4.2/libexec/lib/swipl/library/plunit.pl:1586
ERROR:   [25] translate_message2(plunit(failed(sql,insert2,78,...)),[url(...),':\n\t'|...],_7440) at /opt/homebrew/Cellar/swi-prolog/8.4.2/libexec/lib/swipl/boot/messages.pl:100
ERROR:   [24] translate_message(plunit(failed(sql,insert2,78,...)),[url(...),':\n\t'|...],[]) at /opt/homebrew/Cellar/swi-prolog/8.4.2/libexec/lib/swipl/boot/messages.pl:87
ERROR:   [23] print_message_guarded(error,plunit(failed(sql,insert2,78,...))) at /opt/homebrew/Cellar/swi-prolog/8.4.2/libexec/lib/swipl/boot/messages.pl:1803
ERROR:   [22] setup_call_catcher_cleanup('$messages':push_msg(...),'$messages':print_message_guarded(error,...),_7596,'$messages':pop_msg) at /opt/homebrew/Cellar/swi-prolog/8.4.2/libexec/lib/swipl/boot/init.pl:663
ERROR:   [20] print_message(error,plunit(failed(sql,insert2,78,...))) at /opt/homebrew/Cellar/swi-prolog/8.4.2/libexec/lib/swipl/boot/messages.pl:1772
ERROR:   [18] plunit:failure(sql,insert2,78,error(existence_error(procedure,...),context(...,_7722)),[true(true),...]) at /opt/homebrew/Cellar/swi-prolog/8.4.2/libexec/lib/swipl/library/plunit.pl:1201
ERROR:   [16] plunit:run_test_once(sql,insert2,78,[true(true),...],plunit_sql:'unit body'('insert2@line 78',vars)) at /opt/homebrew/Cellar/swi-prolog/8.4.2/libexec/lib/swipl/library/plunit.pl:842
ERROR:   [14] plunit:run_unit(sql) at /opt/homebrew/Cellar/swi-prolog/8.4.2/libexec/lib/swipl/library/plunit.pl:624
ERROR:   [13] plunit:run_unit_and_check_errors(sql) at /opt/homebrew/Cellar/swi-prolog/8.4.2/libexec/lib/swipl/library/plunit.pl:616
ERROR:   [12] setup_call_catcher_cleanup(plunit:setup_trap_assertions(<clause>(0x600000158000)),plunit:run_unit_and_check_errors(sql),_7892,plunit:report_and_cleanup(<clause>(0x600000158000))) at /opt/homebrew/Cellar/swi-prolog/8.4.2/libexec/lib/swipl/boot/init.pl:663
ERROR:
ERROR: Note: some frames are missing due to last-call optimization.
ERROR: Re-run your program in debug mode (:- debug.) to get more detail.
   Exception: (38) to_list([selec/4], _4454) ? EOF: exit (status 4)
```

while the corresponding error on 8.4.1 was : 

```
ERROR: tests.pl:78:
	test insert2: received error: plunit_sql:'unit body'/2: Unknown procedure: plunit_sql:selec/3
  However, there are definitions for:
        selec/4
```

**Comment by JanWielemaker** [comment]
Thanks.  Fixed using SWI-Prolog/swipl@25c667d54f3337ac70d744813b6563e7b7c40a0d.  Backport issue :cry:

**Comment by kimmens** [comment]
That was fast, thanks Jan :-)
